db/migrate/20150630143001_change_percentage_fields_to_boolean_fields_on_firm.rb

```diff
@@ -0,0 +1,26 @@
+class ChangePercentageFieldsToBooleanFieldsOnFirm < ActiveRecord::Migration
+  BUSINESS_OFFERING_ATTRIBUTE = [
+    :retirement_income_products,
+    :pension_transfer,
+    :long_term_care,
+    :equity_release,
+    :inheritance_tax_and_estate_planning,
+    :wills_and_probate,
+    :other]
+
+  def up
+    BUSINESS_OFFERING_ATTRIBUTE.each do |field|
+      add_column :firms, "#{field}_flag".to_sym, :boolean, null: false, default: false
+      Firm.where("#{field}_percent > ?", 0).update_all("#{field}_flag" => true)
+      remove_column :firms, "#{field}_percent".to_sym, :integer
+    end
+  end
+
+  def down
+    BUSINESS_OFFERING_ATTRIBUTE.each do |field|
+      add_column :firms, "#{field}_percent".to_sym, :integer
+      Firm.where("#{field}_flag" => true).update_all("#{field}_percent" => 100)
```

Comment:
You're right, it doesn't. It seemed like the best thing to do, since the directory only cares about (effectively) truthy or falsey on each field.

The difference would be that in the self-service & signup interface, everything would be either 0 or 100, so actually none of the records would validate based on the pre-change code... which would be a shame, now I think of it.